app/src/main/java/org/glucosio/android/activity/AddA1CActivity.java

```diff
@@ -70,7 +70,7 @@ protected void onCreate(Bundle savedInstanceState) {
         if (this.isEditing()) {
             setTitle(R.string.title_activity_add_hb1ac_edit);
             HB1ACReading readingToEdit = presenter.getHB1ACReadingById(getEditId());
-            readingTextView.setText(readingToEdit.getReading() + "");
+            readingTextView.setText(new StringBuilder().append(this.numberFormat.format(readingToEdit.getReading())).append("").toString());
```

Comment:
This pattern is not required. You don't need to concatenate an empty string. Since you don't concatenate anything then you don't need `StringBuilder` also. So simplification is
```
numberFormat.format(readingToEdit.getReading())
```

The original code is doing `a + ""` to force call `toString()` method that will just print value itself.

Reply:
I was following advice from Android Studio IDE, it suggested the pattern.  There seem to be a lot of warnings / advice from the IDE that we aren't following

Reply:
Yeah, it warns you about `a + ""` which is inefficient and also the bad pattern. But it doesn't warn you about values conversion :)

Reply:
Cris, you can replace all code with builder just with `numberFormat` call

Reply:
Done